index.js

```diff
@@ -502,6 +506,455 @@
   }
   S.is = def('is', {}, [TypeRep(a), $.Any, $.Boolean], is);
 
+  //. ### Showable
+
+  //# toString :: Any -> String
+  //.
+  //. Alias of [`Z.toString`][].
+  //.
+  //. ```javascript
+  //. > S.toString(-0)
+  //. '-0'
+  //.
+  //. > S.toString(['foo', 'bar', 'baz'])
+  //. '["foo", "bar", "baz"]'
+  //.
+  //. > S.toString({x: 1, y: 2, z: 3})
+  //. '{"x": 1, "y": 2, "z": 3}'
+  //.
+  //. > S.toString(S.Left(S.Right(S.Just(S.Nothing))))
+  //. 'Left(Right(Just(Nothing)))'
+  //. ```
+  S.toString = def('toString', {}, [$.Any, $.String], Z.toString);
+
+  //. ### Fantasy Land
+  //.
+  //. Sanctuary is compatible with the [Fantasy Land specification][FL].
+
+  //# equals :: a -> b -> Boolean
+  //.
+  //. Curried version of [`Z.equals`][].
+  //.
+  //. ```javascript
+  //. > S.equals(0, -0)
+  //. false
+  //.
+  //. > S.equals(NaN, NaN)
+  //. true
+  //.
+  //. > S.equals(S.Just([1, 2, 3]), S.Just([1, 2, 3]))
+  //. true
+  //.
+  //. > S.equals(S.Just([1, 2, 3]), S.Just(['1', '2', '3']))
+  //. false
+  //. ```
+  S.equals = def('equals', {}, [a, b, $.Boolean], Z.equals);
+
+  //# concat :: Semigroup a => a -> a -> a
+  //.
+  //. Curried version of [`Z.concat`][].
+  //.
+  //. ```javascript
+  //. > S.concat('abc', 'def')
+  //. 'abcdef'
+  //.
+  //. > S.concat([1, 2, 3], [4, 5, 6])
+  //. [1, 2, 3, 4, 5, 6]
+  //.
+  //. > S.concat({x: 1, y: 2}, {y: 3, z: 4})
+  //. {x: 1, y: 3, z: 4}
+  //.
+  //. > S.concat(S.Just([1, 2, 3]), S.Just([4, 5, 6]))
+  //. Just([1, 2, 3, 4, 5, 6])
+  //. ```
+  S.concat = def('concat', {a: [Z.Semigroup]}, [a, a, a], Z.concat);
+
+  //# empty :: Monoid a => TypeRep a -> a
+  //.
+  //. [Type-safe][sanctuary-def] version of [`Z.empty`][].
+  //.
+  //. ```javascript
+  //. > S.empty(String)
+  //. ''
+  //.
+  //. > S.empty(Array)
+  //. []
+  //.
+  //. > S.empty(Object)
+  //. {}
+  //. ```
+  S.empty = def('empty', {a: [Z.Monoid]}, [TypeRep(a), a], Z.empty);
+
+  //# map :: Functor f => (a -> b) -> f a -> f b
+  //.
+  //. Curried version of [`Z.map`][].
+  //.
+  //. ```javascript
+  //. > S.map(Math.sqrt, [1, 4, 9])
+  //. [1, 2, 3]
+  //.
+  //. > S.map(Math.sqrt, {x: 1, y: 4, z: 9})
+  //. {x: 1, y: 2, z: 3}
+  //.
+  //. > S.map(Math.sqrt, s => s.length)('Sanctuary')
+  //. 3
+  //.
+  //. > S.map(Math.sqrt, S.Just(9))
+  //. Just(3)
+  //.
+  //. > S.map(Math.sqrt, S.Right(9))
+  //. Right(3)
+  //. ```
+  S.map = def('map', {f: [Z.Functor]}, [Fn(a, b), f(a), f(b)], Z.map);
+
+  //# bimap :: Bifunctor f => (a -> b) -> (c -> d) -> f a c -> f b d
+  //.
+  //. Curried version of [`Z.bimap`][].
+  //.
+  //. ```javascript
+  //. > S.bimap(S.toUpper, Math.sqrt, S.Left('foo'))
+  //. Left('FOO')
+  //.
+  //. > S.bimap(S.toUpper, Math.sqrt, S.Right(64))
+  //. Right(8)
+  //. ```
+  S.bimap =
```

Comment:
@svozza `S.either` is for unpacking an `Either a b`, it returns the return value of the mapper. `S.bimap` returns a new `Either` of the value returned by the mapper.